pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go

```diff
@@ -1,20 +1,81 @@
 package certgraphanalysis
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+	corev1 "k8s.io/api/core/v1"
 )
 
-type caBundleRewriteFunc func(caBundle *certgraphapi.CertificateAuthorityBundle)
+const lastRewrittenByAnnotation = "openshift.io/last-rewritten-by"
 
+type configMapRewriteFunc func(configMap *corev1.ConfigMap)
+type secretRewriteFunc func(secret *corev1.Secret)
+type caBundleRewriteFunc func(caBundle *certgraphapi.CertificateAuthorityBundle)
 type certKeyPairRewriteFunc func(certKeyPair *certgraphapi.CertKeyPair)
 
 type metadataOptions struct {
 	rewriteCABundle    caBundleRewriteFunc
 	rewriteCertKeyPair certKeyPairRewriteFunc
+	rewriteConfigMap   configMapRewriteFunc
+	rewriteSecret      secretRewriteFunc
 }
 
 func (metadataOptions) approved() {}
 
+func (l certGenerationOptionList) rewriteCABundle(caBundle *certgraphapi.CertificateAuthorityBundle) {
```

Comment:
why separate these from the definition of type in [s/filtering_options.go](https://github.com/openshift/library-go/pull/1601/files#diff-30028a369ddbe02bd5ade6c03b0f78f1bfab468370e390b854b24b5cb45255d8)